### ISSUE TITLE
TM has missing doc strings and possible param redundancy

### ISSUE THREAD

**Issue by cogmission** [numenta/nupic#3267]
the doc strings for ```maxSegmentsPerCell``` and ```maxSynapsesPerSegment``` seem to be missing. 
see here: https://github.com/numenta/nupic/blob/master/src/nupic/research/temporal_memory.py#L53

and should be entered here: https://github.com/numenta/nupic/blob/master/src/nupic/research/temporal_memory.py#L85 before ```seed```

In addition, I have some confusion as to whether ```maxNewSynapseCount``` should be used alongside ```maxSynapsesPerSegment```, since ```maxNewSynapseCount``` is defined as the maximum segments that can be added during learning and synapses are _**only**_ added during learning - it follows that ```maxNewSynapseCount``` and ```maxSynapsesPerSegment``` may be redundant - and either one or the other should be used?

**Comment by rhyolight** [comment]
@mrcslws Can you take a look?

**Comment by andrewmalta13** [comment]
@cogmission You are right that those two variables are lacking mention in the doc string and that should be addressed.  There is, however, no redundancy between ```maxNewSynapseCount``` and``` maxSynapsesPerSegment```. ``` maxNewSynapseCount``` is the maximum number of synapses that can be created in one call to ```growSynapses```, while ```maxSynapsesPerSegment``` is the maximum number of synapses that can exist on a segment at any given time. 

Since you got confused about this, maybe we change the docstring for ```maxNewSynapseCount``` to read "maximum synapses that can be added during each learning step".

Let me know if that was clear.

**Comment by cogmission** [comment]
@andrewmalta13 
Ah I see. It might be even more clear if you say something like, "The global number of synapses that can be added to one or more segments during each learning step". Because I believe the count is across all segments too?

But yes - that's a lot clearer now. ;-)
